fenics_helpers/plotting/adaptive_plot.py

```diff
@@ -0,0 +1,34 @@
+import matplotlib.pyplot as plt
+
+
+class AdaptivePlot:
+    def __init__(self, fmt="-bx", ax=None):
+        plt.ion()  # interactive on
+        if ax is None:
+            ax = plt.gca()
+
+        self._line, = ax.plot([0.0], [0.0], fmt)
+        self._xs = []
+        self._ys = []
+
+    def __call__(self, x, y):
+        """
+        Add the point (x,y) to the plot and rescales the axes.
+        """
+        self._xs.append(x)
+        self._ys.append(y)
+        self._line.set_data(self._xs, self._ys)
```

Comment:
I don't think the data needs to be stored twice. What about
~~~python
xs, ys = self._line.get_data()
self._line.set_data(np.append(xs, x), np.append(ys, y))
~~~